common/maven-url-handler/src/test/java/io/helidon/build/maven/url/HandlerTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.build.maven.url;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.jar.JarOutputStream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsNull.notNullValue;
+
+public class HandlerTest {
+
+    private static final String CONTENT = "Archetype content";
+    private static final String ARCHIVE_DIRECTORY = "io/helidon/archetypes/helidon-archetype/3.0.0-SNAPSHOT";
+    private static final String ZIP_NAME = "helidon-archetype-3.0.0-SNAPSHOT.zip";
+    private static final String JAR_NAME = "helidon-archetype-3.0.0-SNAPSHOT.jar";
+
+    static Path workDir;
+
+    @BeforeAll
+    static void bootstrap() throws IOException {
+        URL.setURLStreamHandlerFactory(new MavenURLHandlerFactory());
```

Comment:
This is impacting the whole JVM and can have side effects. There should be a way to configure url handler automatically, e.g. system property or class-path.